piet/src/render_context.rs

```diff
@@ -48,11 +50,14 @@ impl ImageFormat {
 /// can implement this trait.
 ///
 /// Code that draws graphics will in general take `&mut impl RenderContext`.
-pub trait RenderContext {
+pub trait RenderContext
+where
+    Self::Brush: IBrush<Self>,
```

Comment:
🤨

Reply:
Can you expand? It turns out we need to write a fair amount of Chalk to get all these types to work together.

Reply:
This is just another instance of "I did not know this would compile". Cool that we can express these sorts of circular relationships though!